### ISSUE TITLE
NullReferenceException on get_SharedDomainAddress onlinux

### ISSUE THREAD

**Issue by skybook888** [microsoft/clrmd#243]
using the gcroot sample by 
```
https://github.com/Microsoft/dotnet-samples/blob/master/Microsoft.Diagnostics.Runtime/CLRMD/GCRoot
```
my only change is change the LoadCrashDump to LoadCoreDump
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Diagnostics.Runtime.Desktop.DesktopRuntimeBase.get_SharedDomainAddress()
   at Microsoft.Diagnostics.Runtime.Desktop.DesktopGCHeap.LoadAllTypes()
   at Microsoft.Diagnostics.Runtime.Desktop.DesktopGCHeap.EnumerateTypes()+MoveNext()
   at Microsoft.Diagnostics.Runtime.Desktop.DesktopGCHeap.EnumerateRoots(Boolean enumerateStatics)+MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at EEHeap.Program.FillRootDictionary(ClrHeap heap) in /coretrace/gcroot/Program.cs:line 175
   at EEHeap.Program.Main(String[] args) in /coretrace/gcroot/Program.cs:line 50
```
i try this on both 2.2 and 3.0 privew container

**Comment by janvorli** [comment]
It is probably due to the fact that the shared domain was removed from .NET core few months ago and daccess now reports the sharedDomain as being null.

**Comment by leculver** [comment]
This was fixed in cd072107.
